### PR TITLE
[nifly] new port

### PR DESCRIPTION
**Pull request by Guekka**
Add port for [nifly](https://github.com/ousnius/nifly/)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
I have only tested Windows 64 and Linux, but all others should work. I disabled support for UWP as I have no idea if this platform is supported

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
Yes

### CODE REVIEW

File: ports/nifly/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,24 @@
+vcpkg_fail_port_install(ON_TARGET "UWP")
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO ousnius/nifly
+    REF 680a3c0435ff78f8189c46de410d0c8aa9c18a06
+    SHA512 b1bb4a194f95e7bcd66280e1b796d0813cd0b4dc292dae5b168092c2d529f89c966a0f11b364787dfa906556181326b30517b7f734fe85246cde4016e3a1d55b
+    HEAD_REF master
```
Comment (srgbtl) [inline ports/nifly/portfile.cmake:8]:
```suggestion
    HEAD_REF main
```

File: ports/nifly/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	SOURCE_PATH "${SOURCE_PATH}"
+	OPTIONS
+		-DBUILD_TESTING=OFF
+)
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(CONFIG_PATH "cmake/")
+
+file(REMOVE_RECURSE
+	${CURRENT_PACKAGES_DIR}/debug/include
+	${CURRENT_PACKAGES_DIR}/debug/share
```
Comment (JonLiu1993) [inline ports/nifly/portfile.cmake:21]:
```suggestion
	"${CURRENT_PACKAGES_DIR}/debug/include"
	"${CURRENT_PACKAGES_DIR}/debug/share"

```

File: ports/nifly/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "nifly",
+  "version-semver": "1.0.0",
```
Comment (JonLiu1993) [inline ports/nifly/vcpkg.json:3]: @Guekka ,Thanks for your pr, could you please clarify why is "version-semver"? does version follow semantic control?

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/nifly/vcpkg.json (hunk 3 above)
Comment (Guekka) [reply ports/nifly/vcpkg.json:3]: It doesn't. I saw "1.0.0" and assumed it did

**Review by JonLiu1993 — APPROVED** [review]

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the new port!

**Comment by Guekka** [comment]
Sure. Mind explaining why vcpkg_fail_port_install` is not necessary?

**Comment by BillyONeal** [comment]
> Sure. Mind explaining why `vcpkg_fail_port_install` is not necessary?

The "supports" field now does that:

```
PS C:\Dev\vcpkg> .\vcpkg.exe install nifly:x64-uwp
Computing installation plan...
Error: nifly[core] is only supported on '!uwp'
PS C:\Dev\vcpkg>
```

**Pull request merged**